### ISSUE TITLE
Web Crash on Release/Product

### ISSUE THREAD

**Issue by AliKales** [flutter/flutter#141782]
Hi, I have developed an app and published it please check it and see the error on console;
[Pinlogy](https://pinlogy.web.app)

As you can see on console there is an error;

`RangeError (end): Invalid value: Not in inclusive range 5..22: -1`

Although there is NO error when i debug it on VS Code.

` sdk: '>=3.2.3 <4.0.0'`

Packages used on project;
`caroby: ^0.1.2
  go_router: ^13.0.1
  easy_localization: ^3.0.3
  encrypt: ^5.0.3
  hive_flutter: ^1.1.0
  json_annotation: ^4.8.1
  firebase_core: ^2.24.2
  firebase_auth: ^4.16.0
  cloud_firestore: ^4.14.0
  url_strategy: ^0.2.0
  flutter_svg: ^2.0.9
  crypto: ^3.0.3
  url_launcher: ^6.2.3`

`[√] Flutter (Channel stable, 3.16.8, on Microsoft Windows [Version 10.0.19045.3930], locale tr-TR)
[√] Windows Version (Installed version of Windows is version 10 or higher)
[√] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
[√] Chrome - develop for the web
[√] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.5.1)
[√] Android Studio (version 2021.2)
[√] VS Code (version 1.85.1)
[√] Connected device (3 available)
[√] Network resources`

**Comment by iapicca** [comment]
without code a reproducible code sample it's impossible to know if the error is caused by your implementation
(or by your own package "caroby")

**Comment by AliKales** [comment]
I have tested my own package and it didnt cause any error. My own package has only extensions and some core UIs so i dont think it is the one causing the problem. If it did then why cant i see that on debug? I really dont get this. There litereally zero error or warning during debug but only after "flutter build web". 
This is a whole project so i dont know what part of code i can show you because console doesnt even tell the current location of problem :/// I'm open to any solution

**Comment by danagbemava-nc** [comment]
Hi @AliKales, do you only face this issue when you deploy your app to a hosting site? If not, can you try running your app in profile mode to see if you get the same issue? (If you do experience the issue in profile mode as well, the stacktrace will not be minified). 

What do you do on startup in your application? 

You can also try integrating an error reporting service like sentry for your webapp to see if it captures a clearer stacktrace which would help with investigating the issue. 

As pointed out above, without a minimal sample, it will be difficult for us to investigate this issue. 

Thank you

**Comment by AliKales** [comment]
Hi @danagbemava-nc, i kept debugging whole time with `flutter run --profile` and i kept getting same error. I tried many things but in the end i removed go_router and wrote my own router with core Dart codes. Right now it works as it should. I like to use go_router but for this project i will continue with core routing. **Anyone who faces this issue i suggest you to remove go_router**! Thank you both for commenting!

**Comment by AliKales** [comment]
Hi again! I noticed my mistake here so i wanted to reopen this issue to let people know actually what was wrong with my code.
Anyone who is facing this issue it is most probably because of go_router but **you don't need to remove go_router from your project!**

As you know we set GoRoute s in GoRouter's routes list. I had written a code that does this automatically but apparently that code crushed and did not let app load in first place.

**So long story short; Check your go router and see if everything is OK!**

Thanks!
